src/test/java/org/javarosa/xform/parse/ExternalSecondaryInstanceParseTest.java

```diff
@@ -134,6 +134,59 @@ public void xformParseException_whenItemsetConfiguresValueOrLabelNotInExternalIn
         }
     }
 
+    @Test
+    public void itemsetBindingVerification_doesNotVerifySecondItem() throws IOException, XFormParser.ParseException {
+        configureReferenceManagerCorrectly();
+
+        Scenario.init("Some form", html(
+            head(
+                title("Some form"),
+                model(
+                    mainInstance(t("data id=\"some-form\"",
+                        t("first")
+                    )),
+
+                    t("instance id=\"mixed-schema\" src=\"jr://file/mixed-schema.xml\""),
+
+                    bind("/data/first").type("string")
+                )
+            ),
+            body(
+                // Define a select using value and label references that only exist for the first item
+                select1Dynamic("/data/first", "instance('mixed-schema')/root/item", "name", "label")
+            )));
+    }
+
+    @Test
+    public void itemsetBindingVerification_verifiesFirstItem() throws IOException {
+        configureReferenceManagerCorrectly();
+
+        try {
+            Scenario.init("Some form", html(
+                head(
+                    title("Some form"),
+                    model(
+                        mainInstance(t("data id=\"some-form\"",
+                            t("first")
+                        )),
+
+                        t("instance id=\"mixed-schema\" src=\"jr://file/mixed-schema2.xml\""),
```

Comment:
Can we use inline instances here or would they not get verified? If we can, they'd make the test much easier to read in my opinion.

Reply:
Coming back at you from the past!

In https://github.com/getodk/javarosa/commit/847865c78512af9b65211ea332d23e17c119fa07#diff-e045a71eed0a5d3fd849d1ca17f98246297bcae6d2d6b62203e17ec297365b03R348, I changed the logic to only verify itemset bindings for external secondary instances. I think that was a mistake so thanks for getting me to investigate.

I have changed the condition so all itemset bindings are verified. I've added test coverage for internal secondary instances. There was actually already some coverage for external secondary instances at https://github.com/getodk/javarosa/blob/a5ea1e95f322165e95e22ff01ea0495fbcfed296/src/test/java/org/javarosa/xform/parse/ExternalSecondaryInstanceParseTest.java#L108